daemon/cdi.go

```diff
@@ -0,0 +1,38 @@
+package daemon
+
+import (
+	"github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+	"github.com/docker/docker/errdefs"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var cdiRegistry cdi.Registry
+
+// initializeCdi initializes the CDI registry and registers the CDI device driver.
+func initializeCdi(d *Daemon) error {
+	features := d.Features()
+	if features == nil {
+		return nil
+	}
+
+	if cdiEnabled, ok := (*features)["cdi"]; !ok || !cdiEnabled {
+		return nil
+	}
+
+	if !d.HasExperimental() {
+		return errdefs.InvalidParameter(errors.New("the cdi featuer is only supported in experimental mode"))
```

Comment:
Typo: "featuer".

Also, unless this is the idiomatic failure for enabling experimental features in non-experimental mode, I think `errdefs.NotImplemented` is more-correct? It mentions "not available due to config" in the description.

Reply:
I was basing this on the handling of `checkpointing` [here](https://github.com/moby/moby/blob/2cf63891427d68661bb8e3c58969058c62dbc85d/daemon/start.go#L21), but you're right that this is not due to a paramter being passed in. I will change to `NotImplemented` instead.